contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -211,10 +234,16 @@ static int _cb_ruby_wait(trilogy_sock_t *sock, trilogy_wait_t wait)
     }
 
     int fd = trilogy_sock_fd(sock);
-    if (rb_wait_for_single_fd(fd, wait_flag, timeout) <= 0)
-        return TRILOGY_SYSERR;
+    if (ctx->io == Qnil) {
+        ctx->io = rb_io_fdopen(fd, 0, "trilogy_socket");
+    }
 
-    return 0;
+    VALUE result = rb_io_wait(ctx->io, RB_INT2NUM(wait_flag), rb_fiber_scheduler_make_timeout(timeout));
```

Comment:
I acknowledge it's what's already happening on sufficiently-new rubies (along with the IO construction), but `rb_fiber_scheduler_make_timeout` sounds like it might allocate. Is that true? Something we can/should avoid?

Reply:
There is a small chance I the future it might allocate but for the moment it doesn't, it just embeds a float into the VALUE. I hope at some point in the future to migrate to fixed point timeouts.

Reply:
Bonus question: is this going to affect our min ruby version (which I note we don't seem to currently declare)? I'm guessing this is moderately recent.

Reply:
No I'll add feature detection, this PR is just what I could manage in the 30 minutes I had before bed time.

Reply:
> it just embeds a float into the VALUE

Flonums only work for a subset of all double, if it doesn't fit it will be an allocation.

Reply:
I think it also always allocate on 32bits?